pandas/io/formats/style.py

```diff
@@ -1028,14 +1046,18 @@ def _background_gradient(s, cmap="PuBu", low=0, high=0, text_color_threshold=0.4
             raise ValueError(msg)
 
         with _mpl(Styler.background_gradient) as (plt, colors):
-            smin = s.values.min()
-            smax = s.values.max()
+            smin = s.min() if vmin is None else vmin
+            if isinstance(smin, ABCSeries):
+                smin = smin.min()
+            smax = s.max() if vmax is None else vmax
+            if isinstance(smax, ABCSeries):
+                smax = smax.max()
             rng = smax - smin
             # extend lower / upper bounds, compresses color range
             norm = colors.Normalize(smin - (rng * low), smax + (rng * high))
             # matplotlib colors.Normalize modifies inplace?
             # https://github.com/matplotlib/matplotlib/issues/5427
-            rgbas = plt.cm.get_cmap(cmap)(norm(s.values))
+            rgbas = plt.cm.get_cmap(cmap)(norm(s.to_numpy(dtype=float)))
```

Comment:
OK with this but can just use default constructor without `dtype` argument I think? As it it seems like this is trying to coerce, which I'm not sure why that would be needed

Reply:
unfortunately, `.to_numpy()` alone will produce `dtype=object` and rejected by matplotlib.
OTOH, it will immediately convert to float anyway since `cmap` require normalizing to [0,1], nothing to loose for coercing in the first place.